models/doctype/Payment/PaymentServer.js

```diff
@@ -113,18 +114,28 @@ export default class PaymentServer extends BaseDocument {
       if (outstandingAmount == null) {
         outstandingAmount = baseGrandTotal;
       }
-      if (this.amount <= 0 || this.amount > outstandingAmount) {
+      if (this.amount.lte(0) || this.amount.gt(outstandingAmount)) {
         let message = frappe._(
-          `Payment amount (${this.amount}) should be less than Outstanding amount (${outstandingAmount}).`
+          `Payment amount: ${frappe.format(
+            this.amount,
+            'Currency'
+          )} should be less than Outstanding amount: ${frappe.format(
+            outstandingAmount,
+            'Currency'
+          )}.`
         );
-        if (this.amount <= 0) {
-          const amt = this.amount < 0 ? ` (${this.amount})` : '';
+
+        if (this.amount.lte(0)) {
+          const amt = this.amount.lt(0)
+            ? `: ${frappe.format(this.amount, 'Currency')}`
+            : '';
```

Comment:
Seems like with pesa the ternary comparison wont be required.

```suggestion

        if (this.amount.lte(0)) {
          const amt = ${frappe.format(this.amount, 'Currency')};
```